<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8' />
    <meta http-equiv="X-UA-Compatible" content="chrome=1" />
    <meta name="description" content="Cushingacromegalystudy : The source code for the cushing and acromegaly studies, currently ongoing" />

    <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">

    <title>Cushingacromegalystudy</title>
  </head>

  <body>

    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/BridgesLab/CushingAcromegalyStudy">View on GitHub</a>

          <h1 id="project_title">Cushingacromegalystudy</h1>
          <h2 id="project_tagline">The source code for the cushing and acromegaly studies, currently ongoing</h2>

            <section id="downloads">
              <a class="zip_download_link" href="https://github.com/BridgesLab/CushingAcromegalyStudy/zipball/master">Download this project as a .zip file</a>
              <a class="tar_download_link" href="https://github.com/BridgesLab/CushingAcromegalyStudy/tarball/master">Download this project as a tar.gz file</a>
            </section>
        </header>
    </div>

    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
        <h1>
<a name="code-and-raw-data-for-acromegaly-and-cushing-analyses" class="anchor" href="#code-and-raw-data-for-acromegaly-and-cushing-analyses"><span class="octicon octicon-link"></span></a>Code and Raw Data for Acromegaly and Cushing Analyses</h1>

<p>This data was analysed at first on the mbni and felix servers to generate counts tables.  The rest of the analysis was performed locally once the counts table was generated.  See the <strong>processing</strong> folder for the code used in the generation of these alignments and counts tables.</p>

<h2>
<a name="data-files" class="anchor" href="#data-files"><span class="octicon octicon-link"></span></a>Data Files</h2>

<p>Data files are located in the <strong>data</strong> directory
The raw data in this analysis is located in <strong>data/raw</strong> and is the following files:</p>

<ul>
<li>
<strong>patient_table.csv</strong> contains the measured parameters for these patients.</li>
<li>
<strong>patient_legend.txt</strong> describes the units used for measurements in the patient table.</li>
<li>
<strong>patient_sample_mapping.csv</strong> maps the patients to their corresponding samples.</li>
<li>
<strong>transcript_counts_table.csv</strong> has the transcript level counts table.</li>
<li>
<strong>exon_counts_table.csv</strong> has the exon level counts table.  These data has not yet been incorporated into our analysis.</li>
<li>
<strong>acromegaly_patient_IGF1.csv</strong> has the IGF-1 levels for the acromegaly patients.</li>
</ul><p>Data files generated by these scripts are located in the <strong>data/processsed</strong> directory.</p>

<h2>
<a name="script-files" class="anchor" href="#script-files"><span class="octicon octicon-link"></span></a>Script Files</h2>

<p>Script files are saved in <strong>scripts</strong> folder and were analysed in this order</p>

<h3>
<a name="counts_table_filteringrmd" class="anchor" href="#counts_table_filteringrmd"><span class="octicon octicon-link"></span></a>counts_table_filtering.Rmd</h3>

<p>This file filters the counts table to show only the most abundant transcript.  It starts with the file <strong>data/raw/transcript_counts_table.csv</strong> and then ends up with <strong>data/processed/filtered_transcript_counts_table.csv.</strong></p>

<h3>
<a name="deseq_analysis_outlierrmd" class="anchor" href="#deseq_analysis_outlierrmd"><span class="octicon octicon-link"></span></a>deseq_analysis_outlier.Rmd</h3>

<p>This file performs the DESeq analysis both including and removing the one outlier patient who was accidentally included in the analysis.  This script takes the files <strong>data/processed/filtered_transcript_counts_table.csv.</strong> and <strong>data/raw/patient_sample_mapping.csv</strong> and generages annoted DESeq results files for both cushing and acromegaly, as well as lists of statistically significant genes.</p>

<h3>
<a name="goseq-analysisrmd" class="anchor" href="#goseq-analysisrmd"><span class="octicon octicon-link"></span></a>goseq-analysis.Rmd</h3>

<p>This script searches KEGG and GO for enriched categories and pathways.  This generates a variety of Gene Ontology Analysis files and the data for Table 2.</p>

<h3>
<a name="heatmapsrmd" class="anchor" href="#heatmapsrmd"><span class="octicon octicon-link"></span></a>heatmaps.Rmd</h3>

<p>This generates the heatmap used in Figure 2.</p>

<h2>
<a name="barplotsrmd" class="anchor" href="#barplotsrmd"><span class="octicon octicon-link"></span></a>barplots.Rmd</h2>

<p>This script generates all the barplots used in Figures 3, 4 and Supplementary Figures 1 and 2.</p>

<h3>
<a name="acromegaly_clinical_analysisrmd" class="anchor" href="#acromegaly_clinical_analysisrmd"><span class="octicon octicon-link"></span></a>acromegaly_clinical_analysis.Rmd</h3>

<p>This script analyses the clinical characteristics of the groups, generating the table for Table 1 and Figure 1.</p>

<h3>
<a name="geo_comparasonsrmd" class="anchor" href="#geo_comparasonsrmd"><span class="octicon octicon-link"></span></a>GEO_comparasons.Rmd</h3>

<p>This script compares our significant genes to another similar dataset.  This set treated adipocytes with growth hormone in vitro and did microarrays after 48h of treatment.</p>

<h3>
<a name="igf_analysisrmd" class="anchor" href="#igf_analysisrmd"><span class="octicon octicon-link"></span></a>igf_analysis.Rmd</h3>

<p>This takes measured IGF-1 levels from acromegaly patients and compares it to <em>IGF1</em>  expression from WAT explants.</p>

<h2>
<a name="figures" class="anchor" href="#figures"><span class="octicon octicon-link"></span></a>Figures</h2>

<p>The figures generated for the manuscript, via the running of these scripts are in the <strong>figures</strong> directory.  These figures are modified for final publication in the <strong>manuscript</strong> folder using Adobe Illustrator CS6.</p>

<h2>
<a name="manuscript" class="anchor" href="#manuscript"><span class="octicon octicon-link"></span></a>Manuscript</h2>

<p>The manuscript files, including the manuscript, the figures, tables and supplementary data are in the <strong>manuscript</strong> directory.  Within this directory are the files generated for uploading the raw and processed data to the Gene Expression omnibus (in the folder <strong>manuscript/GEO_submission</strong>).</p>
      </section>
    </div>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <p class="copyright">Cushingacromegalystudy maintained by <a href="https://github.com/BridgesLab">BridgesLab</a></p>
        <p>Published with <a href="http://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

              <script type="text/javascript">
            var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
            document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
          </script>
          <script type="text/javascript">
            try {
              var pageTracker = _gat._getTracker("UA-41854373-3");
            pageTracker._trackPageview();
            } catch(err) {}
          </script>


  </body>
</html>
