---
title: Annotation of the GRE Dataset
author: "Innocence Harvey and Dave Bridges"
date: "February 23, 2015"
output:
  html_document:
    keep_md: yes
  pdf_document:
    keep_tex: yes
---

```{r global_options, include=FALSE}
library(knitr)
opts_chunk$set(fig.path='figures/',
               echo=FALSE, warning=FALSE, message=FALSE,dev=c('png','pdf'))
```

This data is from the GR CHIP-seq paper by Reddy \textit{et al.}.  They only provided peaks with a p-value cutoff of < 1.0 * 10-10.

```{r data-entry}
datafile <- '../data/raw/Genomic determination of the GC response-Dataset1.txt'
data <- read.table(datafile, skip=12, sep='\t', header=T)
```

The dataset from was downloaded as `r datafile`.

```{r annotation}
#put data in genomic ranges format
library(GenomicRanges)
library(ChIPpeakAnno)
data(TSS.human.NCBI36)

data.ranges <- makeGRangesFromDataFrame(data, keep.extra.columns=T)
annotatedPeak.data.all = annotatePeakInBatch(data.ranges, AnnotationData=TSS.human.NCBI36)
annotatedPeak.data.diff = annotatePeakInBatch(data.ranges[data.ranges$FDR...<5,], AnnotationData=TSS.human.NCBI36)

library(biomaRt)
ensembl = useMart("ENSEMBL_MART_ENSEMBL",
   dataset="hsapiens_gene_ensembl",
   host="may2009.archive.ensembl.org",
   archive=FALSE)

annotation <- getBM(attributes=c('hgnc_symbol','ensembl_gene_id','start_position','end_position'), 
  filters='ensembl_gene_id',
  values=annotatedPeak.data.all$feature,
  mart=ensembl)

annotated.data.all <- merge(as.data.frame(annotatedPeak.data.all), annotation, by.x='feature', by.y='ensembl_gene_id')
annotated.data.all$insideFeature <- as.factor(annotated.data.all$insideFeature)
annotated.data.all$hgnc_symbol <- as.factor(annotated.data.all$hgnc_symbol)
annotated_GR_peaks_file <- '../data/processed/Annotated GR Peaks.csv'
write.csv(annotated.data.all,annotated_GR_peaks_file)

annotated.data.diff <- merge(as.data.frame(annotatedPeak.data.diff), annotation, by.x='feature', by.y='ensembl_gene_id')
annotated.data.diff$insideFeature <- as.factor(annotated.data.diff$insideFeature)
annotated.data.diff$hgnc_symbol <- as.factor(annotated.data.diff$hgnc_symbol)
annotated_GR_peaks_file_sig <- '../data/processed/Annotated GR Peaks - Significantly Changed.csv'
write.csv(annotated.data.diff,annotated_GR_peaks_file_sig)
```

The peaks were annotated by Nearest Start codon using ChIPpeakAnno to hg18.  The distrubution of peak to nearest start is shown below below.  The distributions of significantly differentially occupied sites to all occupied sites was not different by Two-sample Kolmogorov-Smirnov test (p=`r ks.test(annotated.data.all$distancetoFeature, annotated.data.diff$distancetoFeature)$p.value`).

We annotated `r dim(annotated.data.all)[1]` sites from `r length(levels(annotated.data.all$hgnc_symbol))` different genes.  `r dim(annotated.data.diff)[1]` sites from `r length(levels(annotated.data.diff$hgnc_symbol))` different genes were differentially occupied in the presence of dexamethasone.  The annotated data is written out to the files `r annotated_GR_peaks_file` and `r annotated_GR_peaks_file_sig`.

```{r start-site-distance}
par(mfrow=c(1,2))
with(annotated.data.all, plot(density(distancetoFeature), las=1, main="Average distance to nearest TSS"))
with(annotated.data.diff, plot(density(distancetoFeature), las=1, main="Average distance to nearest TSS"))
```

# Session Information
```{r sessionInfo}
sessionInfo()
```