Comparing HTseq_counts with Rsamtool counts
========================================================

```{r Read data, echo=FALSE}
htseq_counts_filename <- "../data/raw/htseq_gene_counts.txt"
deseq_filename <- "../data/processed/acromegaly_deseq_results_no_outlier.csv"
filtered_counts_filename <- "../data/processed/filtered_transcript_counts_table.csv"
filtered_counts <- read.csv(filtered_counts_filename, row.names="X")
htseq_counts <- read.table(htseq_counts_filename, header=TRUE, sep="\t")
deseq_results <- read.csv(deseq_filename, row.names="X")
```

```{r annotation}
library(biomaRt,quietly=TRUE)
#get transcript info matching transcript id to external (HNGC) id
ensembl <- useMart("ensembl",dataset = 'hsapiens_gene_ensembl')
transcriptInfo_noExon <- getBM(
    attributes=c('ensembl_transcript_id','external_gene_id', 'strand',
                 'chromosome_name', 'ensembl_gene_id',
                 'transcript_start', 'transcript_end'), 
    values = rownames(filtered_counts),
    mart=ensembl)
annotated_counts_noExon <- merge(transcriptInfo_noExon, filtered_counts, by.x="ensembl_transcript_id", by.y="row.names")
annotated_deseq_results <- merge(transcriptInfo_noExon, deseq_results, by.x="ensembl_transcript_id", by.y="id")
htseq_merge_filtered_counts <- merge(annotated_counts_noExon, htseq_counts, by.x="ensembl_gene_id", by.y="Genes")
```

```{r calculate the relative difference}
#rsamtools counts - htseq counts
diff <- abs(htseq_merge_filtered_counts[,8:31] - htseq_merge_filtered_counts[,32:55]) #Rsamtools counts - htseq counts
#calculate relative difference
rel_diff<-diff/htseq_merge_filtered_counts[,8:31] #divide by the Rsamtools counts
rel_diff$average<-rowMeans(rel_diff)
rownames(rel_diff) <- htseq_merge_filtered_counts$ensembl_gene_id
```

```{r plot expression vs relative difference}
counts_merge_exp <- merge(rel_diff, annotated_deseq_results, by.x="row.names", by.y="ensembl_gene_id")
#counts_merge_exp.subset <- counts_merge_exp[counts_merge_exp$baseMeanB < 6000,]
pdf('figure/HtSeq_Rsamtools counts relative difference.pdf')
plot(counts_merge_exp$baseMeanB, counts_merge_exp$average, main="Htseq and Rsamtools Relative Difference vs Avg Exp ", xlab="Acromegaly Avg Expression", ylab="Avg Relative Difference", xlim=range(0:200000), ylim=range(0:3), pch=19)
dev.off()
```
Session Information
---------------------

```{r session-information}
sessionInfo()
```