---
title: "BMIasConfounder"
author: "Quynh Tran"
date: "July 24, 2014"
output: html_document
---

This is to check if BMI is a confouding factor between disease and gene expressions.

```{r read files, echo=FALSE}
sample_file <- "../data/raw/patient_sample_mapping.csv"
patient_file <- "../data/raw/patient_table.csv"

sample_mapping <-read.csv(sample_file)
patient_info <- read.csv(patient_file)

mapping <- merge(sample_mapping, patient_info, by.x="patient..", by.y="id")
```

```{r exclusion, echo=FALSE}
remove_patient <- 29
```

The counts tables were generated previously using  **HTseq.sh** shell script and **merge.command** on Hera.  It removes the outlier data point from the analysis, which was patient **`r remove_patient`**.
This script requires a transcript counts table.  There is also a sample mapping file called **patient_sample_mapping.csv** which links the diagnosis the the samples.  This file was most recently processed on ```r date()```.

```{r loading-data, echo=FALSE, message=FALSE, warning=FALSE}
#required packages
library(DESeq2 ,quietly=TRUE) #for deseq analysis
transcript.counts <- read.table("../data/raw/htseq_gene_counts_GRCh37.74.txt", header=TRUE)
rownames(transcript.counts) <- transcript.counts[,1]
transcript.counts <- transcript.counts[,-1]

#set the conditions
#mapping <- read.csv("../data/raw/patient_sample_mapping.csv")
mapping$samplename <- paste("sample", mapping$sample.., sep="")
mapping$group <- relevel(mapping$group, ref="non-functioning")
#levels(mapping$group) <- c("Control", "Acromegaly", "Cushing's")
levels(mapping$group)[levels(mapping$group)=="non-functioning"] <- "Control"
levels(mapping$group)[levels(mapping$group)=="Cushing's"] <- "Cushing's"
levels(mapping$group)[levels(mapping$group)=="acromegaly"] <- "Acromegaly"

#this command removes the cushings samples
acromegaly.mapping <- droplevels(mapping[mapping$group != "Cushing's" & mapping$patient!=remove_patient,])
acromegaly.counts <- transcript.counts[,acromegaly.mapping$samplename]

#this command removes the acromegaly samples
cushing.mapping <- droplevels(mapping[mapping$group != "Acromegaly"&mapping$samplename %in% colnames(transcript.counts),])
cushing.counts <- transcript.counts[,cushing.mapping$samplename]

#break bmi into normal, overweight, obese
acromegaly.mapping$BMI1 <- cut(acromegaly.mapping$BMI, breaks=c(0,25,30,50))
cushing.mapping$BMI1 <- cut(cushing.mapping$BMI, breaks=c(0,30,50))
cushing.mapping$BMI.cat <- ifelse(cushing.mapping$BMI1=="(0,30]", "Not_obese", "Obese")
cushing.mapping$BMI.cat <- factor(cushing.mapping$BMI.cat)

```

This step gets the protein coding genes only.

```{r protein coding genes, echo=FALSE, message=FALSE}
#required packages
library(biomaRt,quietly=TRUE)
#download gene name annotation data for results sets.
ensembl <- useMart("ensembl",dataset = 'hsapiens_gene_ensembl')
transcriptInfo <- getBM(
    attributes=c('external_gene_id', 'ensembl_gene_id'), 
    filter= 'biotype',
    values = 'protein_coding',
    mart=ensembl)

cushing.protein.coding <- cushing.counts[rownames(cushing.counts)%in%transcriptInfo$ensembl_gene_id,]
acromegaly.protein.coding <- acromegaly.counts[rownames(acromegaly.counts)%in%transcriptInfo$ensembl_gene_id,]


```

```{r cds-construction, echo=FALSE, message=FALSE}
acromegaly.mapping$condition <- factor(paste0(acromegaly.mapping$group, ".",acromegaly.mapping$BMI1))
#acromegaly.mapping <- acromegaly.mapping[order(acromegaly.mapping$group),]
cds.acromegaly.bmi = DESeqDataSetFromMatrix(countData=acromegaly.protein.coding, colData=acromegaly.mapping, design=~BMI1 + group + BMI1:group)
acromegaly.mapping$group<- relevel(acromegaly.mapping$group, "Control")
acromegaly.mapping$BMI1<- relevel(acromegaly.mapping$BMI1, "(0,25]")


#group is the variable of interest, so put group at the end of the design formula
#cushing.mapping$condition <- factor(paste0(cushing.mapping$group, ".",cushing.mapping$BMI))
#cushing.mappingsub <- cushing.mapping[cushing.mapping$BMI1 %in% c("(0,30]","(30,50]"),]
cushing.mapping$BMIxGroup <- factor(paste0(cushing.mapping$BMI.cat, ".", cushing.mapping$group))
cushing.counts <- cushing.protein.coding[,colnames(cushing.protein.coding) %in% cushing.mapping$samplename]
cds.cushing.bmi = DESeqDataSetFromMatrix(countData=cushing.counts, colData=cushing.mapping, design=~BMI.cat+group+BMI.cat:group)
cds.cushing.bmiGroup = DESeqDataSetFromMatrix(countData=cushing.counts, colData=cushing.mapping, design=~BMIxGroup)
#cushing.mappingsub$BMI1<- droplevels(cushing.mappingsub$BMI1)
cushing.mapping$BMI.cat <- relevel(cushing.mapping$BMI.cat, "Not_obese")
cushing.mapping$group <- relevel(cushing.mapping$group, "Control")
```
Full Analysis
--------------

```{r deseq-analysis adjusting for age, echo=FALSE, dev=c('pdf','png'), message=FALSE}
#### ACROMEGALY: has many comparisons, not yet finish#########
acromegaly.cds.bmi <- DESeq(cds.acromegaly.bmi)
plotMA(acromegaly.cds.bmi,ylim=c(-2,2),main="DESeq2 - Acromegaly adjusting for BMI")
acromegaly.results.bmi <- results(acromegaly.cds.bmi)
sum(acromegaly.results.bmi$padj<.05, na.rm=TRUE)
resultsNames(acromegaly.cds.bmi)
#getting the reuslts for bmi as an factor variable
'acro.disease.normal <- results(acromegaly.cds.bmi, contrast=list("BMI.30.50..groupAcromegaly", "BMI.0.25..groupAcromegaly"))
sum(acro.disease.normal$padj<.05, na.rm=TRUE)
acro.disease.overweight <- results(acromegaly.cds.bmi, contrast=list("BMI.25.30..groupAcromegaly", "BMI.25.30..groupControl"))
acro.disease.obese <- results(acromegaly.cds.bmi, contrast=list("BMI.30.50..groupAcromegaly", "BMI.30.50..groupControl"))
#acro.bmi.combined <- merge(acromegaly.results.bmi, acro.bmi.contrast, by.x="row.names", by.y="row.names")
acro.bmi.combined <- cbind(acro.disease.normal, acro.disease.overweight)
colnames(acro.bmi.combined)[2:7]<- gsub("x","AcrovsControl", colnames(acro.bmi.combined)[2:7])
colnames(acro.bmi.combined)[8:13]<-gsub("y","Obese_vs_Overweight", colnames(acro.bmi.combined[8:13]))
acro.bmi.combined <- acro.bmi.combined[order(acro.bmi.combined$padj.bmi, na.last=TRUE),]
#acromegaly.results.bmi <- acromegaly.results.bmi[order(acromegaly.results.bmi$padj, na.last=TRUE),]
write.csv(acro.bmi.combined, "../data/processed/acromegaly_results_GRCh37.74-adjusting_bmi.csv")'

###### CUSHING'S #########
#************To look at the interaction term only, use the model that has all group, bmi, and groupxbmi ***************
cushing.cds.bmi <- DESeq(cds.cushing.bmi)
plotMA(cushing.cds.bmi,ylim=c(-2,2),main="DESeq2 - Cushing adjusting for BMI")
cushing.results.bmi <- results(cushing.cds.bmi)
sum(cushing.results.bmi$padj<0.05, na.rm=TRUE)
resultsNames(cushing.cds.bmi)

cushing.results.bmi$log2FC_Obese_vs_notObese <- results(cushing.cds.bmi, name="BMI.cat_Obese_vs_Not_obese")$log2FoldChange
cushing.results.bmi$lfcSE_Obese_vs_notObese <- results(cushing.cds.bmi, name="BMI.cat_Obese_vs_Not_obese")$lfcSE
cushing.results.bmi$pvalue_Obese_vs_notObese <- results(cushing.cds.bmi, name="BMI.cat_Obese_vs_Not_obese")$pvalue
cushing.results.bmi$padj_Obese_vs_notObese <- results(cushing.cds.bmi, name="BMI.cat_Obese_vs_Not_obese")$padj

cushing.results.bmi$log2FC_Cushing_vs_Con <- results(cushing.cds.bmi, name="group_Cushing.s_vs_Control")$log2FoldChange
cushing.results.bmi$lfcSE_Cushing_vs_Con <- results(cushing.cds.bmi, name="group_Cushing.s_vs_Control")$lfcSE
cushing.results.bmi$pvalue_Cushing_vs_Con <- results(cushing.cds.bmi, name="group_Cushing.s_vs_Control")$pvalue
cushing.results.bmi$padj_Cushing_vs_Con <- results(cushing.cds.bmi, name="group_Cushing.s_vs_Control")$padj

cushing.results.bmi <- cushing.results.bmi[order(cushing.results.bmi$padj, na.last=TRUE),]
#write.csv(cushing.results.bmi, "../data/processed/cushing_results_GRCh37.74-adjusting_bmi.csv")

#### To look at genes that are affected by Cushing's in the Not obese or Obese group, I used the model that has the new variable bmiGroup#######
cushing.cds.bmixgroup <- DESeq(cds.cushing.bmiGroup)
cushing.results.bmixgroup <- results(cushing.cds.bmixgroup)
plotMA(cushing.cds.bmixgroup,ylim=c(-2,2),main="DESeq2 - Cushing adjusting for BMI using BMIxGroup variable")

cushing.results.bmixgroup$log2FC_NotObese_CushingvCon <- results(cushing.cds.bmixgroup, contrast=c("BMIxGroup","Not_obese.Cushing.s", "Not_obese.Control"))$log2FoldChange
cushing.results.bmixgroup$lfcSE_NotObese_CushingvCon <- results(cushing.cds.bmixgroup, contrast=c("BMIxGroup","Not_obese.Cushing.s", "Not_obese.Control"))$lfcSE
cushing.results.bmixgroup$pvalue_NotObese_CushingvCon <- results(cushing.cds.bmixgroup, contrast=c("BMIxGroup","Not_obese.Cushing.s", "Not_obese.Control"))$pvalue
cushing.results.bmixgroup$padj_NotObese_CushingvCon <- results(cushing.cds.bmixgroup, contrast=c("BMIxGroup","Not_obese.Cushing.s", "Not_obese.Control"))$padj
sum(cushing.results.bmixgroup$padj_NotObese_CushingvCon<0.05, na.rm=TRUE)

cushing.results.bmixgroup$log2FC_Obese_CushingvCon <- results(cushing.cds.bmixgroup, contrast=c("BMIxGroup","Obese.Cushing.s", "Obese.Control"))$log2FoldChange
cushing.results.bmixgroup$lfcSE_Obese_CushingvCon <- results(cushing.cds.bmixgroup, contrast=c("BMIxGroup","Obese.Cushing.s", "Obese.Control"))$lfcSE
cushing.results.bmixgroup$pvalue_Obese_CushingvCon <- results(cushing.cds.bmixgroup, contrast=c("BMIxGroup","Obese.Cushing.s", "Obese.Control"))$pvalue
cushing.results.bmixgroup$padj_Obese_CushingvCon <- results(cushing.cds.bmixgroup, contrast=c("BMIxGroup","Obese.Cushing.s", "Obese.Control"))$padj
sum(cushing.results.bmixgroup$padj_Obese_CushingvCon<0.05, na.rm=TRUE)



```

```{r graphs, dev='pdf', echo=FALSE}
####reshape####
library(ggplot2)
library(reshape2)
acro.fitted <- assays(acromegaly.cds.bmi)[["mu"]]
fitted.common.scale = t( t( acro.fitted) / sizeFactors(acromegaly.cds.bmi) )
acro.residuals <- counts(acromegaly.cds.bmi, normalized=TRUE) - fitted.common.scale
acromegaly.protein.coding <- cbind(acromegaly.protein.coding, acro.fitted)
acromegaly.protein.coding <- acromegaly.protein.coding[19:36]
acromegaly.protein.coding <- acromegaly.protein.coding[complete.cases(acromegaly.protein.coding),]
acromegaly.long <- reshape(acromegaly.protein.coding, direction="long", 
                           ids=rownames(acromegaly.protein.coding), idvar="Genes",
                           varying=colnames(acromegaly.protein.coding), v.names="Fitted_Values",
                           timevar="Condition", times=acromegaly.mapping$condition)
acromegaly.long1 <- reshape(acromegaly.protein.coding, direction="long", 
                           ids=rownames(acromegaly.protein.coding), idvar="Genes",
                           varying=colnames(acromegaly.protein.coding), v.names="Fitted_Values",
                           timevar="Disease",times=acromegaly.mapping$group)
acromegaly.long2 <- reshape(acromegaly.protein.coding, direction="long", 
                           ids=rownames(acromegaly.protein.coding), idvar="Genes",
                           varying=colnames(acromegaly.protein.coding), v.names="Fitted_Values",
                           timevar="BMI",times=acromegaly.mapping$BMI1)
acromegaly.long$Disease <-acromegaly.long1$Disease
acromegaly.long$BMI <- ifelse(acromegaly.long2$BMI=="(0,25]", "Normal", ifelse(acromegaly.long2$BMI=="(25,30]", "Overweight", "Obsese"))
par(mfrow=c(1,2))
interaction.plot(acromegaly.long$Disease, acromegaly.long$BMI, acromegaly.long$Fitted_Values, main="Acro-InteractionPlot", ylab="Mean of Fitted Values", xlab="Disease")
plot(acro.fitted, acro.residuals, main="Acro-Fitted vs Residuals", xlab="Fitted", ylab="Residuals", pch=19)
dev.copy2pdf(file="figure/Acro_interaction_plot.pdf")

################CUSHING'S###################
cushing.fitted <- assays(cushing.cds.bmi)[["mu"]]
fitted.common.scale = t( t( cushing.fitted) / sizeFactors(cushing.cds.bmi) )
cushing.residuals <- counts(cushing.cds.bmi, normalized=TRUE) - fitted.common.scale
cushing.counts <- cbind(cushing.counts, cushing.fitted)
cushing.counts <- cushing.counts[17:32]
cushing.counts <- cushing.counts[complete.cases(cushing.counts),]
cushing.long1 <- reshape(cushing.counts, direction="long", 
                           ids=rownames(cushing.counts), idvar="Genes",
                           varying=colnames(cushing.counts), v.names="Fitted_Values",
                           timevar="Disease",times=cushing.mapping$group)
cushing.long2 <- reshape(cushing.counts, direction="long", 
                           ids=rownames(cushing.counts), idvar="Genes",
                           varying=colnames(cushing.counts), v.names="Fitted_Values",
                           timevar="BMI",times=cushing.mapping$BMI1)
cushing.residuals.sub <- cushing.residuals[complete.cases(cushing.residuals),]
cushing.residuals.long <- melt(cushing.residuals.sub, value.name="Residuals", variable.name=cushing.mapping$samplename, id.vars="row.names")
cushing.residuals.long$BMI <- cushing.long2$BMI
#cushing.long$Disease <-cushing.long1$Disease
cushing.long1$BMI <- ifelse(cushing.long2$BMI=="(0,30]", "Not Obese", "Obsese")
par(mfrow=c(2,2))
interaction.plot(cushing.long1$Disease, cushing.long1$BMI, cushing.long1$Fitted_Values, main="Cushing-InteractionPlot", xlab="Disease", ylab="Mean of Fitted Values")
plot(cushing.fitted, cushing.residuals, main="Cushing-Fitted vs Residuals", xlab="Fitted", ylab="Residuals", pch=19)
plot(Residuals~BMI, cushing.residuals.long)
dev.copy2pdf(file="figure/Cushing_interaction_plot.pdf")
```


```{r pca, dev='pdf', echo=FALSE}
require(graphics)
cushing.fitted.noNA <- cushing.fitted[complete.cases(cushing.fitted),]
cushing.pca <- prcomp(cushing.fitted.noNA,scale=TRUE)
summary(cushing.pca)
#scree plot
par(mfrow=c(1,2))
plot(cushing.pca, main="Scree Plot")
plot(cushing.pca$x[,1:2], pch=20,  col=c("red","blue", "green" ))
dev.copy2pdf(file= "figure/PCA_2d.pdf" )

#mycolors <- c("red", "green", "blue") # Define plotting colors.
library(scatterplot3d) # Loads library scatterplot3d.
par(mfrow=c(1,1))
scatterplot3d(cushing.pca$x[,1:3], pch=20, highlight.3d=TRUE, main="3d PCA")
dev.copy2pdf(file= "figure/PCA_3d.pdf" )

#PCA of samples
pca <- as.data.frame(prcomp(t(na.omit(counts(cushing.cds.bmi, normalized=TRUE))))$x)
col <- 1:2
groups <- colData(cushing.cds.bmi)$group
plot(PC2~PC1, data=pca, bg=col[groups], pch=20, main="PCA for samples", col=groups)
text(pca$PC1, pca$PC2, rownames(pca), pos= 2)
dev.copy2pdf(file= "figure/PCA_samples.pdf" )
```

Clustering

```{r kmeans, dev='pdf', echo=FALSE}
library(RColorBrewer)
k <- 3
norm.counts <- counts(cushing.cds.bmi, normalized=TRUE)
notAllZero <- (rowSums(norm.counts) > 0)
norm.counts.noNA <- norm.counts[notAllZero,]
km <- kmeans(t(scale(t(norm.counts.noNA))), 3)
niceCols <- brewer.pal(k, "Dark2")
plot(norm.counts.noNA[,"2"],norm.counts.noNA[,"17"],col=niceCols[km$cluster], pch=19)
points(km$centers, col = niceCols[1:k], pch = 8, cex=2)
dev.copy2pdf(file="figure/k-means.pdf")

#library(cluster)
#set.seed(112358)
#dist <- dist(norm.counts.noNA, method="euclidean")
#cl<-hc(dist)
#plot(norm.counts.noNA[,"2"],norm.counts.noNA[,"17"], col=niceCols[cl$cluster],pch=19)
#plot(cl) 
#dev.copy2pdf(file="figure/k-meloids.pdf")

```

Separate the cushing's patients into two groups: obese and not obese to examine the effect of the disease on gene expression when adjusting for age


```{r Cushing effect on obese and not obese patients, echo=FALSE}
#library(reshape2)
######## Separate the cushing's patients into two groups: obese and not obese to examine the effect of the disease on gene expression when adjusting for age

#####NOTE: need to add AGE into the design #################
cushing.mapping.obese <- cushing.mapping[cushing.mapping$BMI1=="(30,50]",] 
cushing.counts.obese <- cushing.protein.coding[,colnames(cushing.protein.coding) %in% cushing.mapping.obese$samplename]
cds.cushing.obese = DESeqDataSetFromMatrix(countData=cushing.counts.obese, colData=cushing.mapping.obese, design=~group)
cushing.cds.obese <- DESeq(cds.cushing.obese)
#plotMA(cushing.cds.obese,ylim=c(-2,2),main="Cushing's effect in obese patients")
cushing.results.obese <- results(cushing.cds.obese)
sum(cushing.results.obese$padj<0.05, na.rm=TRUE)
resultsNames(cushing.cds.obese)

cushing.group.obese <- results(cushing.cds.obese, contrast=c("group", "Cushing.s", "Control"))
cushing.group.obese <- cushing.group.obese[order(cushing.group.obese$padj, na.last=TRUE),]
#write.csv(cushing.group.obese, "../data/processed/cushing_results_Obese.csv")

cushing.fitted <- assays(cushing.cds.obese)[["mu"]]
fitted.common.scale = t( t( cushing.fitted) / sizeFactors(cushing.cds.obese) )
cushing.residuals <- counts(cushing.cds.obese, normalized=TRUE) - fitted.common.scale
cushing.residuals.sub <- cushing.residuals[complete.cases(cushing.residuals),]
cushing.residuals.long <- melt(cushing.residuals.sub, value.name="Residuals", variable.name=cushing.mapping.obese$samplename, id.vars="row.names")
#plot(cushing.fitted, cushing.residuals, main="Cushing-Fitted vs Residuals Obese", xlab="Fitted", ylab="Residuals", pch=19)

################
cushing.mapping.notObese <- cushing.mapping[cushing.mapping$BMI1=="(0,30]",]
cushing.counts.notObese <-cushing.protein.coding[,colnames(cushing.protein.coding) %in% cushing.mapping.notObese$samplename]
cds.cushing.notObese = DESeqDataSetFromMatrix(countData=cushing.counts.notObese, colData=cushing.mapping.notObese, design=~group)
cushing.cds.notObese <- DESeq(cds.cushing.notObese)
#plotMA(cushing.cds.notObese,ylim=c(-2,2),main="DESeq2 - Cushing's effect in NOT obese patients")
cushing.results.notObese <- results(cushing.cds.notObese)
sum(cushing.results.notObese$padj<0.05, na.rm=TRUE)
resultsNames(cushing.cds.notObese)

cushing.group.notObese <- results(cushing.cds.notObese, contrast=c("group", "Cushing.s", "Control"))
cushing.group.notObese <- cushing.group.notObese[order(cushing.group.notObese$padj, na.last=TRUE),]
#write.csv(cushing.group.notObese, "../data/processed/cushing_results_NOT_Obese.csv")

cushing.fitted <- assays(cushing.cds.notObese)[["mu"]]
fitted.common.scale = t( t( cushing.fitted) / sizeFactors(cushing.cds.notObese) )
cushing.residuals <- counts(cushing.cds.notObese, normalized=TRUE) - fitted.common.scale
cushing.residuals.sub <- cushing.residuals[complete.cases(cushing.residuals),]
cushing.residuals.long <- melt(cushing.residuals.sub, value.name="Residuals", variable.name=cushing.mapping.obese$samplename, id.vars="row.names")
#plot(cushing.fitted, cushing.residuals, main="Cushing-Fitted vs Residuals notObese", xlab="Fitted", ylab="Residuals", pch=19)

```

Annotation
-------------

This step annotates the data tables with the official gene symbols.

```{r annotation, echo=FALSE, message=FALSE}
cushing.results.bmixgroup <- cushing.results.bmixgroup[,7:14]
cushing.bmi.combined <- merge(cushing.results.bmi, transcriptInfo, by.x="row.names", by.y="ensembl_gene_id")
cushing.results.bmixgroup <- merge(cushing.results.bmixgroup, transcriptInfo, by.x="row.names", by.y="ensembl_gene_id")
#cushing.group.notObese <- merge(cushing.group.notObese, transcriptInfo, by.x="row.names", by.y="ensembl_gene_id")

#write annotated results files.
write.csv(cushing.bmi.combined, "../data/processed/Annotated_Results_Cushings_bmi_adjusting.csv")
write.csv(cushing.results.bmixgroup, "../data/processed/Annotated_Results_Cushings_NotObese_and_Obese.csv")
#write.csv(cushing.group.obese, "../data/processed/Annotated_Results_Cushings_Obese.csv")
#write.csv(cushing.group.notObese, "../data/processed/Annotated_Results_Cushings_NotObese.csv")
#write.csv(acro.bmi.combined, "../data/processed/Annotated_Results_GRCh37.74_Acromegaly_bmi.csv")
```

Session Information
-------------------

For the R session, the package versions were:
```{r}
sessionInfo()
```