Analysis of Methylation Data from Cushing's Study
=============================================================

This file was last compiled on ```r date()```.

```{r data entry, echo=FALSE, warning=FALSE}
library(xlsx)
methylation.data.file <- '../data/raw/Summary Table of Cushings Relative Control Methylation Jan 6 2013.xlsx'
methylation.data <- read.xlsx2(methylation.data.file, sheetIndex=1,startRow=2)
methylation.data$Average.Beta.Control <- as.numeric(as.character(methylation.data$Average.Beta.Control))
methylation.data$Average.Beta.Cushings <- as.numeric(as.character(methylation.data$Average.Beta.Cushings))
methylation.data$DiffScore.Cushings <- as.numeric(as.character(methylation.data$DiffScore.Cushings))
methylation.data$Delta.Beta.Cushings <- as.numeric(as.character(methylation.data$Delta.Beta.Cushings))

expression.data.file <- '../data/processed/htseq_Annotated DESeq2 Results - Cushing.csv'
expression.data <- read.csv(expression.data.file)
combined.data <- merge(expression.data, methylation.data,
                       by.x='external_gene_id',
                       by.y='Gene.Symbol')
combined.data$Delta.Beta.Cushings <- as.numeric(combined.data$Delta.Beta.Cushings)
combined.data$DiffScore.Cushings <- as.numeric(combined.data$DiffScore.Cushings)
```

The methylation is in `r methylation.data.file` wherea se we used `r expression.data.file` for the relative expression.

Statics
---------

Tested the predictive value on the delta-beta value on the fold change.

```{r methylation-all-gene-plots, dev=c('png','pdf'), echo=FALSE}
lm.fit.tss1500 <- lm(2^log2FoldChange~Delta.Beta.Cushings, data=combined.data[combined.data$UCSC_REFGENE_GROUP=='TSS1500',])
summary(lm.fit.tss1500)
par(mfrow=c(2,2))
plot(lm.fit.tss1500)
lm.fit.tss200 <- lm(2^log2FoldChange~Delta.Beta.Cushings, data=combined.data[combined.data$UCSC_REFGENE_GROUP=='TSS200',])
summary(lm.fit.tss200)
plot(lm.fit.tss200)
```

Are Differentially Methylated Genes More Likely to Be Differentially Expressed?
---------------------------------------------------------------------------------

To test this, we asked whether the genes with a significantly different methylation pattern had a significantly different expression pattern.  We then compared this to a randomly sampled set of genes from the same expression set.  We then repeated this analysis 1000 times for each of the hypo and hyper expressed genes, and tested how often the chis-quared test led to a significant result.

```{r diff-expression, echo=FALSE, warning=FALSE}
hypermethylated.genes <- levels(droplevels(combined.data[combined.data$Delta.Beta.Cushings>0,]$external_gene_id))
hypomethylated.genes <- levels(droplevels(combined.data[combined.data$Delta.Beta.Cushings<0,]$external_gene_id))


old.warn <- options()$warn
options(warn = -1)
hyper.bootstrap <- rep(NA,1000)
for (n in seq(1,1000)){
  hyper.diff.expressors <- table(subset(expression.data, external_gene_id%in%hypermethylated.genes)$padj < 0.05)
  hyper.null.set <- sample(expression.data$external_gene_id, length(hypermethylated.genes), replace=F)
  hyper.null <- table(subset(expression.data, external_gene_id%in%hyper.null.set )$padj < 0.05)
  hyper.bootstrap[n] <- try(chisq.test(hyper.diff.expressors, p=hyper.null, rescale.p=T)$p.value, silent=T)
}
table(hyper.bootstrap < 0.05)

hypo.bootstrap <- rep(NA,1000)
for (n in seq(1,1000)){
  hypo.diff.expressors <- table(subset(expression.data, external_gene_id%in%hypomethylated.genes)$padj < 0.05)
  hypo.null.set <- sample(expression.data$external_gene_id, length(hypomethylated.genes), replace=F)
  hypo.null <- table(subset(expression.data, external_gene_id%in%hypo.null.set )$padj < 0.05)
  hypo.bootstrap[n] <- try(chisq.test(hypo.diff.expressors, p=hypo.null, rescale.p=T)$p.value, silent=T)
}
table(hypo.bootstrap < 0.05)
options(warn = old.warn)
```

Figures
----------

```{r methylation-significant-genes, dev=c('png','pdf'), echo=FALSE, fig.show='asis'}
with(combined.data, plot(Delta.Beta.Cushings,log2FoldChange
                         , pch=19, cex=0.1,
                         col=UCSC_REFGENE_GROUP,
                         main="All Methylation Sites"))
legend("topleft", levels(combined.data$UCSC_REFGENE_GROUP), pch=19, cex=1, bty="n",
       col=palette()[1:6])

with(combined.data[combined.data$UCSC_REFGENE_GROUP=='TSS200',], 
     plot(Delta.Beta.Cushings,2^log2FoldChange,
                         pch=19, cex=0.1,
                         main="Only TSS200"))
abline(lm.fit.tss200)

with(combined.data[combined.data$UCSC_REFGENE_GROUP=='TSS1500',], 
     plot(Delta.Beta.Cushings,2^log2FoldChange,
                       pch=19, cex=0.1,
                         main="Only TSS1500"))
abline(lm.fit.tss1500)

```

How Many Genes were Differentially Methylated
----------------------------------------------

```{r differentially-methylated, echo=FALSE}
hypomethylated.data <- droplevels(subset(methylation.data, Delta.Beta.Cushings<0))
hypermethylated.data <- droplevels(subset(methylation.data, Delta.Beta.Cushings>0))

hypomethylated_genes_file <- '../data/processed/Cushings Hypomethylated Genes.txt'
write(levels(hypomethylated.data$Gene.Symbol), hypomethylated_genes_file)

hypermethylated_genes_file <- '../data/processed/Cushings Hypermethylated Genes.txt'
write(levels(hypermethylated.data$Gene.Symbol), hypermethylated_genes_file)
```

There was `r length(levels(methylation.data$Gene.Symbol))` that had differential methylation including `r length(levels(hypomethylated.data$Gene.Symbol))` hypomethylated genes and `r length(levels(hypermethylated.data$Gene.Symbol))` hypermethylated genes.  There was also `r length(intersect(levels(hypermethylated.data$Gene.Symbol),levels(hypomethylated.data$Gene.Symbol)))` genes which were hypomethylated in one location and hypermethulated in another.  These genes are written to a files named `r hypomethylated_genes_file` and `r hypermethylated_genes_file`.

Only genes in TS1500
---------------------

Session Information
-------------------
```{r session-info}
sessionInfo()
```
