Analysis of Methylation Data from Cushing's Study
=============================================================

This file was last compiled on ```r date()```.

```{r data entry, echo=FALSE}
methylation.data.file <- '../data/raw/Cushings GC Analysis.csv'
methylation.data <- read.csv(methylation.data.file)
expression.data.file <- '../data/processed/Annotated DESeq Results - Cushing.csv'
expression.data <- read.csv(expression.data.file)
combined.data <- merge(expression.data, methylation.data,
                       by.x='external_gene_id',
                       by.y='UCSC_REFGENE_NAME',
                       all.y=T, all.x=F)
```

The methylation is in `r methylation.data.file` wherea se we used `r expression.data.file` for the relative expression.

Statics
---------

Tested the predictive value on the delta-beta value on the fold change.

```{r statistics}
lm.fit.tss1500 <- lm(foldChange~Cushings.Delta.Beta, data=combined.data[combined.data$UCSC_REFGENE_GROUP=='TSS1500',])
summary(lm.fit.tss1500)
par(mfrow=c(2,2))
plot(lm.fit.tss1500)
lm.fit.tss200 <- lm(foldChange~Cushings.Delta.Beta, data=combined.data[combined.data$UCSC_REFGENE_GROUP=='TSS200',])
summary(lm.fit.tss200)
plot(lm.fit.tss200)
```

Are Differentially Methylated Genes More Likely to Be Differentially Expressed?
---------------------------------------------------------------------------------

To test this, we asked whether the genes with a significantly different methylation pattern had a significantly different expression pattern.  We then compared this to a randomly sampled set of genes from the same expression set.  We then repeated this analysis 1000 times for each of the hypo and hyper expressed genes, and tested how often the chis-quared test led to a significant result.

```{r diff-expression, echo=FALSE, warning=FALSE}
hypermethylated.genes <- levels(droplevels(combined.data[combined.data$Cushings.Delta.Beta>0,]$external_gene_id))
hypomethylated.genes <- levels(droplevels(combined.data[combined.data$Cushings.Delta.Beta<0,]$external_gene_id))


old.warn <- options()$warn
options(warn = -1)
hyper.bootstrap <- rep(NA,1000)
for (n in seq(1,1000)){
  hyper.diff.expressors <- table(subset(expression.data, external_gene_id%in%hypermethylated.genes)$padj < 0.05)
  hyper.null.set <- sample(expression.data$external_gene_id, length(hypermethylated.genes), replace=F)
  hyper.null <- table(subset(expression.data, external_gene_id%in%hyper.null.set )$padj < 0.05)
  hyper.bootstrap[n] <- try(chisq.test(hyper.diff.expressors, p=hyper.null, rescale.p=T)$p.value, silent=T)
}
table(hyper.bootstrap < 0.05)

hypo.bootstrap <- rep(NA,1000)
for (n in seq(1,1000)){
  hypo.diff.expressors <- table(subset(expression.data, external_gene_id%in%hypomethylated.genes)$padj < 0.05)
  hypo.null.set <- sample(expression.data$external_gene_id, length(hypomethylated.genes), replace=F)
  hypo.null <- table(subset(expression.data, external_gene_id%in%hypo.null.set )$padj < 0.05)
  hypo.bootstrap[n] <- try(chisq.test(hypo.diff.expressors, p=hypo.null, rescale.p=T)$p.value, silent=T)
}
table(hypo.bootstrap < 0.05)
options(warn = old.warn)
```

Figures
----------

```{r figures, dev=c('png','pdf'), echo=FALSE, fig.show='asis'}
with(combined.data, plot(Cushings.Delta.Beta,foldChange
                         , pch=19, cex=0.1,
                         col=UCSC_REFGENE_GROUP,
                         main="All Methylation Sites"))
legend("topleft", levels(combined.data$UCSC_REFGENE_GROUP), pch=19, cex=1, bty="n",
       col=palette()[1:6])

with(combined.data[combined.data$UCSC_REFGENE_GROUP=='TSS200',], 
     plot(Cushings.Delta.Beta,foldChange,
                         pch=19, cex=0.1,
                         main="Only TSS200"))
abline(lm.fit.tss200)

with(combined.data[combined.data$UCSC_REFGENE_GROUP=='TSS1500',], 
     plot(Cushings.Delta.Beta,foldChange,
                       pch=19, cex=0.1,
                         main="Only TSS1500"))
abline(lm.fit.tss1500)

```
Session Information
-------------------
```{r session-info}
sessionInfo()
```
