DESeq Analysis of Cushing and Acromegaly Patient Samples with Outlier Removed
===============================================================================

```{r exclusion, echo=FALSE}
remove_patient <- 29
```

The counts tables were generated previously using HTseq and a shell script.  It removes the outlier data point from the analysis, which was patient **`r remove_patient`**.
This script requires a transcript counts table.  There is also a sample mapping file called **patient_sample_mapping.csv** which links the diagnosis the the samples.  This file was most recently processed on ```r date()```.

```{r loading-data, echo=FALSE, message=FALSE}
#required packages
library(DESeq ,quietly=TRUE) #for deseq analysis
library(knitcitations) #for generation of citations
write.bibtex(c('base',names(sessionInfo()$otherPkgs)), file = "references.bib", append=T, create_key=F) #writes references to bibtex file
bib <- read.bibtex("references.bib") #loads the bib file


transcript.counts <- read.table("../data/raw/htseq_gene_counts.txt", header=TRUE)
rownames(transcript.counts) <- transcript.counts[,1]
transcript.counts <- transcript.counts[,-1]
#convert all column names to lower case
colnames(transcript.counts) <- tolower(colnames(transcript.counts))
#replace "_" with nothing 
colnames(transcript.counts) <- gsub("_", "", colnames(transcript.counts))
#set the conditions
mapping <- read.csv("../data/raw/patient_sample_mapping.csv")
mapping$samplename <- paste("sample", mapping$sample.., sep="")
mapping$group <- relevel(mapping$group, ref="non-functioning")
levels(mapping$group) <- c("Control", "Acromegaly", "Cushing's")

filtered.mapping <- mapping[mapping$patient.. != 29,]
filtered.transcript.counts <- transcript.counts[,colnames(transcript.counts) != mapping[mapping$patient.. == 29,]$samplename]
```

```{r cds-construction, echo=FALSE, message=FALSE}
#this command matches the mappings to the transcript headers, and extracts the conditions
conditions <- mapping[match(names(filtered.transcript.counts),filtered.mapping$samplename),]$group
cds = newCountDataSet( filtered.transcript.counts, conditions )
```

These data were analysed in  by DESeq `r citep(bib[["DESeq"]])`. We did not remove lower expressing genes because we pre-filtered the data to examine only one transcript per gene.  

Full Analysis
--------------

```{r deseq-analysis, echo=FALSE, message=FALSE}
cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

write.csv(counts(cds, normalized=TRUE ),'../data/processed/Normalized_htseq_Counts_Table.csv')

acromegaly.results = nbinomTest( cds, "Control", "Acromegaly" )
write.csv(acromegaly.results, "../data/processed/acromegaly_deseq_results_with_htseq.csv")

cushing.results = nbinomTest( cds, "Control", "Cushing's" )
write.csv(cushing.results, "../data/processed/cushing_deseq_results_with_htseq.csv")
```



Annotation
-------------

This step annotates the data tables with the official gene symbols.

```{r annotation, echo=FALSE, message=FALSE}
#required packages
library(knitcitations, quietly=TRUE) #for generation of citations
library(biomaRt,quietly=TRUE)
library(plyr) #for using the ddply function
write.bibtex(c('base',names(sessionInfo()$otherPkgs)), file = "references.bib", append=T, create_key=F) #writes references to bibtex file
bib <- read.bibtex("references.bib") #loads the bib file
#download gene name annotation data for results sets.
ensembl <- useMart("ensembl",dataset = 'hsapiens_gene_ensembl')
transcriptInfo <- getBM(
    attributes=c('external_gene_id', 'ensembl_gene_id'), 
    values = 'ensembl_gene_id',
    mart=ensembl)
#merge results files with transcript info files.
annotated.cushing.results <- merge(cushing.results, transcriptInfo, by.x="id", by.y="ensembl_gene_id")
annotated.acromegaly.results <- merge(acromegaly.results, transcriptInfo, by.x="id", by.y="ensembl_gene_id")

#write annotated results files.
write.csv(annotated.cushing.results, "../data/processed/htseq_Annotated DESeq Results - Cushing.csv")
write.csv(annotated.acromegaly.results, "../data/processed/htseq_Annotated DESeq Results - Acromegaly.csv")
```

The data was annotated from Ensembl data, using the biomaRt package `r citep(bib[[c("biomaRt1","biomaRt2")]])`.

Differentially Expressed Genes
--------------------------------
```{r differentially-expressed, dev=c('png','pdf'), echo=FALSE, fig.show='asis'}
acromegaly.diff.genes <- subset(annotated.acromegaly.results, padj <= 0.05) 
with(acromegaly.diff.genes, plot(log2FoldChange, padj, pch=16, main="Differentially Expressed Genes in Acromegaly Patients Using htseq counts", xlim=range(0:10), ylim=range(0:1)))
acromegaly.up.genes <- subset(acromegaly.diff.genes, foldChange > 1)
write(as.character(acromegaly.up.genes), file = "../data/processed/Acromegaly Upregulated Genes_htseq.txt")

acromegaly.down.genes <- subset(acromegaly.diff.genes, foldChange < 1)
write(as.character(acromegaly.down.genes), file = "../data/processed/Acromegaly Downregulated Genes_htseq.txt")

cushing.diff.genes <- subset(annotated.cushing.results , padj <= 0.05) 
with(cushing.diff.genes, plot(log2FoldChange, padj, pch=16, main="Differentially Expressed Genes in Cushing's Patients using htseq counts"))
cushing.up.genes <- subset(cushing.diff.genes, foldChange > 1)
write(as.character(cushing.up.genes), file = "../data/processed/Cushing Upregulated genes_htseq.txt")

cushing.down.genes <- subset(cushing.diff.genes, foldChange < 1)
write(as.character(cushing.down.genes), file = "../data/processed/Cushing Downregulated Genes_htseq.txt")
```

### Acromegaly

There were **`r dim(acromegaly.diff.genes)[1]`** differentially expressed genes from the acromegaly patients, with **`r table(acromegaly.diff.genes$log2FoldChange > 0)[1]`** genes downregulated and **`r table(acromegaly.diff.genes$log2FoldChange > 0)[2]`** genes upregulated.

### Cushing's Disease

There were **`r dim(cushing.diff.genes)[1]`** differentially expressed genes from the acromegaly patients, with **`r table(cushing.diff.genes$log2FoldChange > 0)[1]`** genes downregulated and **`r table(cushing.diff.genes$log2FoldChange > 0)[2]`** genes upregulated.


Bibiography
------------
```{r bibliography, echo=FALSE, results='asis'}
bibliography("html") 
```

Session Information
-------------------

For the R session, the package versions were:
```{r}
sessionInfo()
```
