Analysis of Data for Acromegaly Patients by Heatmaps
=============================================================

Statistics
----------

```{r data-entry, echo=FALSE}
#for references
require(knitcitations)
bib <- read.bibtex("references.bib") #loads the bib file

#load in the previously processed and annoted acromegaly analysis results
analysis.file <- "../data/processed/Annotated DESeq Results without Outlier - Acromegaly.csv"
analysis.data <- read.csv(analysis.file)
transcripts.file <- "../data/processed/Normalized Counts Table.csv"
transcript.counts <- read.csv(transcripts.file, row.names="X")
#set the conditions
mapping <- read.csv("../data/raw/patient_sample_mapping.csv")
mapping$samplename <- paste("sample", mapping$sample.., sep="")
mapping$group <- relevel(mapping$group, ref="non-functioning")
levels(mapping$group) <- c("Control", "Acromegaly","Cushing's")

filtered.mapping <- mapping[mapping$patient.. != 29,]
filtered.transcript.counts <- transcript.counts[,colnames(transcript.counts) != mapping[mapping$patient.. == 29,]$samplename]

#subset out differentially expressed transcripts
diff.transcripts <- droplevels(subset(analysis.data, padj <= 0.05))
```
This file was most recently processed on ```r date()```.  This uses the genes which were subsetted (only the highest 40\%).  Also, this uses the DESeq normalized data.

Data Clustering
----------------
The following is a cluster analysis of the scaled transcript counts.  This was done with bootstrap resampling of the normalized, scaled transcript counts using the **pvclust** package`r citep(bib['pvclust'])`). 

```{clustering, echo=FALSE, dev=c('png','pdf'), echo=FALSE, fig.show='asis'}
#scaled the data
scaled.sample.transcript.counts <- scale(filtered.transcript.counts)

#cluster number determination (see http://www.statmethods.net/advstats/cluster.html)
wss <- (nrow(scaled.sample.transcript.counts)-1)*sum(apply(scaled.sample.transcript.counts,2,var))
for (i in 2:15) wss[i] <- sum(kmeans(scaled.sample.transcript.counts, 
     centers=i)$withinss)
plot(1:15, wss, type="b", xlab="Number of Clusters",
  ylab="Within groups sum of squares",
     main="Cluster Number Determination (3)")

library(pvclust)
fit <- pvclust(scaled.sample.transcript.counts, method.hclust="ward",
   method.dist="euclidean")
plot(fit) # dendogram with p values
# add rectangles around groups highly supported by the data
pvrect(fit, alpha=.95)
```


Differentially Expressed Genes
----------------------------------

To test the grouping of differentially expressed transcripts, we only examined genes with significantly different transcripts based on DESeq analysis.

```{r de-heatmap, dev=c('png','pdf'), echo=FALSE, fig.show='asis', warning=FALSE}

#subset for differenially expressed transcripts
differential.transcripts <- diff.transcripts$id
diff.exprs <- filtered.transcript.counts[as.character(differential.transcripts),]

#reordered based on fold change
ordered.transcripts <- diff.transcripts[order(diff.transcripts$foldChange),]
diff.exprs.ordered <- diff.exprs[as.character(ordered.transcripts$id),]
#show only unique genes
diff.exprs.ordered.genes <- diff.exprs.ordered[!(duplicated(ordered.transcripts$external_gene_id, incomparables=F)),]
rownames(diff.exprs.ordered.genes) <- as.character(ordered.transcripts[!(duplicated(ordered.transcripts$external_gene_id, incomparables=F)),]$external_gene_id)


library(gplots,quietly=TRUE)
library(RColorBrewer,quietly=TRUE)
bars <- brewer.pal(2, "Set1")
color.map <- function(x) { if (x=="Acromegaly") bars[1] else bars[2] }

mapping.acromegaly <- subset(filtered.mapping, group != "Cushing's")
diagnosis.colors <- sapply(mapping.acromegaly[order(match(mapping.acromegaly$samplename,colnames(diff.exprs))),]$group, FUN=color.map)

#remove cushing samples
diff.exprs.acromegaly <- diff.exprs.ordered.genes[,colnames(diff.exprs.ordered.genes) %in% mapping.acromegaly$samplename]

plot(as.dendrogram(hclust(dist(t(scale(as.matrix(diff.exprs.acromegaly)))))))
heatmap.colors <- brewer.pal(11, "BrBG")
heatmap.2(log(as.matrix(diff.exprs.acromegaly)+1), scale="row", Rowv=F, ColSideColors=diagnosis.colors, col=heatmap.colors, density.info="none", trace="none", labCol=F, margins=c(6,6))

#write it out to a final figure as well
pdf('../figures/differential-expressor-heatmap.pdf')
heatmap.colors <- brewer.pal(11, "BrBG")
heatmap.2(log(as.matrix(diff.exprs.acromegaly)+1), scale="row", Rowv=F, ColSideColors=diagnosis.colors, col=heatmap.colors, density.info="none", trace="none", labCol=F, margins=c(6,6))
dev.off()
```

References
-----------
```{r bibliography, echo=FALSE, results='asis'}
bibliography()
```

Session Information
-------------------
```{r session-info}
sessionInfo()
```
