Analysis of Data for Acromegaly Patients by Heatmaps
=============================================================

Statistics
----------

```{r data-entry, echo=FALSE}
#for references
require(knitcitations)
bib <- read.bibtex("references.bib") #loads the bib file

#load in the previously processed and annoted acromegaly analysis results
analysis.file <- "../data/processed/Annotated Results GRCh37.74 - Acromegaly.csv"
analysis.data <- read.csv(analysis.file, row.names='X')
transcripts.file <- "../data/processed/RPKM_VSD_counts_table_Acromegaly.csv"
transcript.counts <- read.csv(transcripts.file, row.names="X")
#set the conditions
mapping <- read.csv("../data/raw/patient_sample_mapping.csv")
mapping$samplename <- paste("sample", mapping$sample.., sep="")
mapping$group <- relevel(mapping$group, ref="non-functioning")
levels(mapping$group) <- c("Control", "Acromegaly","Cushing's")

filtered.mapping <- mapping[mapping$patient.. != 29,]
filtered.transcript.counts <- transcript.counts[,colnames(transcript.counts) != mapping[mapping$patient.. == 29,]$samplename]

#subset out differentially expressed transcripts
diff.transcripts <- droplevels(subset(analysis.data, padj <= 0.05))
```
This file was most recently processed on ```r date()```.  This uses the DESeq normalized data.


Differentially Expressed Genes
----------------------------------

To test the grouping of differentially expressed transcripts, we only examined genes with significantly different transcripts based on DESeq analysis.

```{r de-heatmap, dev=c('png','pdf'), echo=FALSE, fig.show='asis', warning=FALSE}

#subset for differenially expressed transcripts
differential.transcripts <- rownames(diff.transcripts)
diff.exprs <- filtered.transcript.counts[differential.transcripts %in% rownames(filtered.transcript.counts),]

#reordered based on fold change
ordered.transcripts <- diff.transcripts[order(diff.transcripts$log2FoldChange),]
diff.exprs.ordered <- diff.exprs[rownames(ordered.transcripts),]
#show only unique genes
diff.exprs.ordered.genes <- diff.exprs.ordered[!(duplicated(ordered.transcripts$external_gene_id, incomparables=F)),]
rownames(diff.exprs.ordered.genes) <- as.character(ordered.transcripts[!(duplicated(ordered.transcripts$external_gene_id, incomparables=F)),]$external_gene_id)


library(gplots,quietly=TRUE)
library(RColorBrewer,quietly=TRUE)
bars <- brewer.pal(2, "Set1")
color.map <- function(x) { if (x=="Acromegaly") bars[1] else bars[2] }

mapping.acromegaly <- subset(filtered.mapping, group != "Cushing's")
diagnosis.colors <- sapply(mapping.acromegaly[order(match(mapping.acromegaly$samplename,colnames(diff.exprs))),]$group, FUN=color.map)

#remove cushing samples
diff.exprs.acromegaly <- diff.exprs.ordered.genes[,colnames(diff.exprs.ordered.genes) %in% mapping.acromegaly$samplename]

plot(as.dendrogram(hclust(dist(t(scale(as.matrix(diff.exprs.acromegaly)))))))
heatmap.colors <- brewer.pal(11, "BrBG")
heatmap.2(log(as.matrix(diff.exprs.acromegaly)+1), scale="row", Rowv=F, ColSideColors=diagnosis.colors, col=heatmap.colors, density.info="none", trace="none", labCol=F, margins=c(6,6))
dev.copy(pdf, '../figures/differential-expressor-heatmap-RPKM.pdf')
dev.off()

#write it out to a final figure as well
#pdf('../figures/differential-expressor-heatmap-RPKM.pdf')
#heatmap.colors <- brewer.pal(11, "BrBG")
#heatmap.2(log(as.matrix(diff.exprs.acromegaly)+1), scale="row", Rowv=F, ColSideColors=diagnosis.colors, col=heatmap.colors, #density.info="none", trace="none", labCol=F, margins=c(6,6))
#dev.off()
```

References
-----------
```{r bibliography, echo=FALSE, results='asis'}
bibliography()
```

Session Information
-------------------
```{r session-info}
sessionInfo()
```
