Title making the RPKM table
========================================================

Making the RPKM Table

```{r Read data, echo=FALSE}
filtered_counts_filename <- "../data/processed/filtered_transcript_counts_table.csv"
filtered_counts <- read.csv(filtered_counts_filename, row.names="X")
processed_filename <- "../data/processed/RPKM_counts_table.csv"

#Divide each column (sample) by the total number of reads (sum of that column)
rpm_table <- sweep(filtered_counts, 2, colSums(filtered_counts), '/')
rpm_table <- sweep(rpm_table, 2, 10^9, '*')
```

```{r annotation, echo=FALSE}
library(biomaRt,quietly=TRUE)
#get transcript info matching transcript id to external (HNGC) id
ensembl <- useMart("ensembl",dataset = 'hsapiens_gene_ensembl')
transcriptInfo <- getBM(
    attributes=c('ensembl_transcript_id','external_gene_id', 'strand',
                 'chromosome_name', 'ensembl_gene_id',
                 'transcript_start', 'transcript_end',
                 'start_position', 'end_position'), 
    values = rownames(filtered_counts),
    mart=ensembl)
annotated_counts <- merge(transcriptInfo, rpm_table, by.x="ensembl_transcript_id", by.y="row.names")

#making the transcript length column by taking transcript_end - transcript_start
annotated_counts$TLength <- annotated_counts[,7] - annotated_counts[,6]

```

```{r Getting RPKM: reads per kilobase per million reads }
#Divide each row of the rpm_table by the transcript length
rpkm_table <- annotated_counts[,c(10:33)]/annotated_counts$TLength
rpkm_table_withRownames <- data.frame(rpkm_table, row.names=annotated_counts$ensembl_transcript_id)
rpkm_table_withRownames$geneName <- annotated_counts$external_gene_id

write.csv(rpkm_table_withRownames, file=processed_filename)

```

```{r}
#rpkm_table <- RPKM(filtered_counts, feature.size=transcript.size.withRownames, lib.size=colSums(filtered_counts))
```
Session Information
---------------------

```{r session-information}
sessionInfo()
```

